src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -166,13 +167,15 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         val element = ref.reference.resolve() as? RsVisible ?: return
         val oMod = o.contextStrict<RsMod>() ?: return
         if (element.isVisibleFrom(oMod)) return
-
+        val withinOneCrate = element.crateRoot == (o as RsElement).crateRoot
```

Comment:
Let's change the type of `o` to `RsElement` in the function signature and use `o.crateRoot`

Reply:
Thanks, fixed.

Reply:
Great! Don't forget to move fixes creations to `prepare` methods (see the unresolved comment).